### PR TITLE
fix(priority): Add priority and first_release to fields that trigger GroupAttributes updates

### PR DESCRIPTION
**Pull request by snigdhas**
The snapshots to kafka are produced with the post_save signal if the updated fields match a preset list. We added priority and first_release to GroupAttributes but never extended the list of supported fields for a snapshot. This PR fixes that in order to fix the snuba search [bug](https://sentry.sentry.io/feedback/?referrer=slack&notification_uuid=a89100e3-90f1-409e-88cf-19cd8d8d7cf7&alert_rule_id=15210908&alert_type=issue&feedbackSlug=javascript%3A5983986729&project=11276) where manual updates to priority were not reflected in the search results. 

We'll need to re-backfill group attributes since any manual updates will not have been captured by these snapshots.

### CODE REVIEW

**Review by JoshFerge — APPROVED** [review]
classic. signals delenda est. nice catch!

is there any way we could write a test and do a snuba search more e2e? no worries if not

File: src/sentry/issues/priority.py [hunk 1]
```diff
@@ -51,7 +51,8 @@ def update_priority(
         return
 
     previous_priority = PriorityLevel(group.priority) if group.priority is not None else None
-    group.update(priority=priority)
+    group.priority = priority
+    group.save()
```
[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/issues/priority.py:55]: `group.update` sends post save signals https://github.com/getsentry/sentry/blob/646521b0759c9b369ab65332cb2767cc573b653a/src/sentry/db/models/query.py#L107-L113

Are you sure this is the root cause of the problem?

[the remark this answers is not in the document]
Comment (snigdhas) [reply src/sentry/issues/priority.py]: ahh I thought I'd seen that code somewhere but couldn't find it and didn't have `groups.enable-post-update-signal` enabled locally. thanks!

[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/issues/priority.py]: `groups.enable-post-update-signal` is for `Group.objects.filter(id).update`. That's different to `group.update(priority=new_priority)`, which is basically doing the same thing that save does

File: src/sentry/issues/attributes.py [hunk 2]
```diff
@@ -252,7 +252,7 @@ def process_update_fields(updated_fields) -> set[str]:
         # we'll need to assume any of the attributes are updated in that case
         updated_fields = {"all"}
     else:
-        VALID_FIELDS = {"status", "substatus", "num_comments"}
+        VALID_FIELDS = {"status", "substatus", "num_comments", "priority", "first_release"}
```
Comment (wedamija) [inline src/sentry/issues/attributes.py:255]: Ahh I see, this makes more sense!

**Review by wedamija — APPROVED** [review]

**Pull request merged**